xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-api/src/main/java/org/xwiki/security/authorization/DefaultAuthorizationManager.java

```diff
@@ -173,9 +176,17 @@ private boolean evaluateSecurityAccess(Right right, DocumentReference userRefere
 
     @Override
     public Right register(RightDescription rightDescription) throws UnableToRegisterRightException
+    {
+        // By default Admin should imply any newly registered right.
+        return register(rightDescription, Collections.singleton(Right.ADMIN));
```

Comment:
PROGRAM too.

Reply:
ok adding it

Reply:
Note that the current logic is to add Admin (and Program) only if the list of impliedBy right is not given. If a list is given I consider that the developer chose to not specify Admin/Program right. That sounds good to you?

Reply:
Hmm not sure. Feels weird to have a right which is not implied by PROGRAM, I'm pretty sure people will forget to add PROGRAM and ADMIN when they want their right to be in implied by EDIT.

Reply:
So you suggest to always force adding Program and Admin in register?

Reply:
It should definitely be force for PROGRAM since that's kind of the definition of PROGRAM. Admin I'm not sure.

Reply:
Forcing them both for now.